### PR TITLE
Added RegFile for SIMD&FP registers and instruction support for LDRDui, STRDui and ADDv8i8

### PR DESCRIPTION
**Pull request by tanmaytirpankar**
Please look at the commit messages of individual commits.
Test status same as before these changes except for two additional tests part of the commit which also pass.

### CODE REVIEW

**Review by regehr — CHANGES_REQUESTED** [review]
thanks again for doing this!! hopefully these are easy

File: backend_tv/adjust_src.cpp [hunk 1]
```diff
@@ -302,10 +302,6 @@ void checkSupport(Instruction &i) {
       *out << "\nERROR: llvm GC instrinsics not supported\n\n";
       exit(-1);
     }
-    if (callee.find("llvm.objc") != string::npos) {
```
[the remark this answers is not in the document]
Comment (tanmaytirpankar) [reply backend_tv/adjust_src.cpp:305]: I saw there were two blocks if blocks doing the same thing. Redundant code.

[the remark this answers is not in the document]
Comment (regehr) [reply backend_tv/adjust_src.cpp:305]: ah, now I see. thanks :)

File: backend_tv/lifter.cpp [hunk 2]
```diff
@@ -745,10 +810,36 @@ class arm2llvm {
     Value *V = nullptr;
     if (op.isImm()) {
       V = getIntConst(op.getImm(), size);
-    } else {
+    } else if (op.isReg()) {
       V = readFromReg(op.getReg());
       if (size == 32)
         V = createTrunc(V, getIntTy(32));
+    } else {
+      auto expr = op.getExpr();
```
[the remark this answers is not in the document]
Comment (tanmaytirpankar) [reply backend_tv/lifter.cpp:818]: Yes. I wanted to ask about this. You have created a map called GOT for dealing with :got: relocation specifier. But we can create a common map dealing with all relocation specifiers. I can create a function to do this and replace GOT.

[the remark this answers is not in the document]
Comment (regehr) [reply backend_tv/lifter.cpp:818]: yes, please do that.

so far I did not put a lot of thought into dealing with globals, so it's not very well designed!

File: backend_tv/adjust_src.cpp [hunk 3]
```diff
@@ -297,15 +289,6 @@ void checkSupport(Instruction &i) {
       *out << "\nERROR: llvm.thread instrinsics not supported\n\n";
       exit(-1);
     }
-    if ((callee.find("llvm.experimental.gc") != string::npos) ||
```
Comment (regehr) [inline backend_tv/adjust_src.cpp:300]: wait, I don't think this code was duplicated!

**Review by regehr — CHANGES_REQUESTED** [review]
looks great besides the one thing

File: backend_tv/adjust_src.cpp (hunk 3 above)
Comment (tanmaytirpankar) [reply backend_tv/adjust_src.cpp:300]: Ah my bad. I'll fix it. I may have accidentally deleted that one.

**Pull request merged**